packages/shared-ux/avatar/user_profile/impl/user_profiles_selectable.tsx

```diff
@@ -0,0 +1,317 @@
+                !selectedOptionsToAdd.find((option) => option.key === profile.uid)
+            )
+            .map(toSelectableOption)
+        : [];
+
+      // Merge in any new options and add group separator if necessary
+      if (defaultOptionsToAdd.length) {
+        const separatorIndex = ensureSeparator(nextOptions);
+        nextOptions.splice(separatorIndex, 0, ...selectedOptionsToAdd);
+        nextOptions.push(...defaultOptionsToAdd);
+      } else {
+        nextOptions.push(...selectedOptionsToAdd);
+      }
+
+      return nextOptions;
+    });
+  };
+
+  // Marks displayed options as checked or unchecked depending on `props.selectedOptions`
+  const updateCheckedStatus = () => {
+    setDisplayedOptions((values) =>
+      values.map((option) => {
+        if (selectedOptions) {
+          const match = selectedOptions.find((p) => p.uid === option.key);
+          return { ...option, checked: match ? 'on' : undefined };
+        }
+        return { ...option, checked: undefined };
+      })
+    );
+  };
+
+  useEffect(resetDisplayedOptions, [options]); // eslint-disable-line react-hooks/exhaustive-deps
+  useEffect(updateDisplayedOptions, [defaultOptions, selectedOptions]); // eslint-disable-line react-hooks/exhaustive-deps
+  useEffect(updateCheckedStatus, [options, defaultOptions, selectedOptions]);
+
+  const selectedCount = selectedOptions ? selectedOptions.length : 0;
+
+  return (
+    <EuiSelectable
+      options={displayedOptions}
+      // @ts-expect-error: Type of `nextOptions` in EuiSelectable does not match what's actually being passed back so need to manually override it
+      onChange={(nextOptions: Array<EuiSelectableOption<{ data: UserProfileWithAvatar }>>) => {
+        if (!onChange) {
+          return;
+        }
+
+        // Take all selected options from `nextOptions` unless already in `props.selectedOptions`
+        const values: UserProfileWithAvatar[] = nextOptions
+          .filter((option) => {
+            if (option.isGroupLabel || option.checked !== 'on') {
+              return false;
+            }
+            if (selectedOptions && selectedOptions.find((p) => p.uid === option.key)) {
+              return false;
+            }
+            return true;
+          })
+          .map((option) => option.data);
+
+        // Add all options from `props.selectedOptions` unless they have been deselected in `nextOptions`
+        if (selectedOptions && !singleSelection) {
+          selectedOptions.forEach((profile) => {
+            const match = nextOptions.find((o) => o.key === profile.uid);
+            if (!match || match.checked === 'on') {
+              values.push(profile);
+            }
+          });
+        }
+
+        onChange(values);
+      }}
+      style={{ maxHeight: height }}
+      singleSelection={singleSelection}
+      searchable
+      searchProps={{
+        placeholder:
+          searchPlaceholder ??
+          i18n.translate(
```

Comment:
should this be a constant?

Reply:
I dig it - [702fb2e](https://github.com/elastic/kibana/pull/139247/commits/702fb2e4725f60d47cc2989e440b98053b042a7f) for updates